### ISSUE TITLE
Feature Request: Let run.sh pass arguments to redis

### ISSUE THREAD

**Issue by seyhbold** [bitnami/bitnami-docker-redis#90]
Hi,

could you append a ` $@` to the lines

https://github.com/bitnami/bitnami-docker-redis/blob/b55c4c408d412dcced971f6052d2b201333e1e7e/4.0/rootfs/run.sh#L15

and

https://github.com/bitnami/bitnami-docker-redis/blob/b55c4c408d412dcced971f6052d2b201333e1e7e/4.0/rootfs/run.sh#L17

So that we can pass parameters down to redis?

Thank you very much

**Comment by tompizmor** [comment]
Hello @seyhbold,

Could you please add an example of how you plan to start the container? 

This feature is already supported in the mariadb via an environment variable. Do you think the same approach would be also valid for redis? Check the following example:

https://github.com/bitnami/bitnami-docker-mariadb#passing-extra-command-line-flags-to-mysqld-startup

Tomas

**Comment by seyhbold** [comment]
Seems reasonable to me.
I want to use the image through the redis helm chart and configure it as a cache
So i want to pass --maxmemory 100mb --maxmempolicy ...

So in the first attempt I replaced /run.sh and it worked.
But later I figured that I would need to provide my own redis.conf mainly to enable flushdb commands again. And found no way to do that with this image.

Each time I copied or mounted a conf file to /bitnami/redis/conf nami_initialize complained.
What am I doing wrong?

**Comment by tompizmor** [comment]
@seyhbold 
> So in the first attempt I replaced /run.sh and it worked.

Yes, that should work. At the moment you can just build your own docker image changing the run.sh script and adding whatever flag in [this line](https://github.com/bitnami/bitnami-docker-redis/blob/master/4.0/rootfs/run.sh#L8)

> Each time I copied or mounted a conf file to /bitnami/redis/conf nami_initialize complained.
What am I doing wrong?

Could you please describe the steps to reproduce the issue? Can you also paste the error that you get?

**Comment by seyhbold** [comment]
Sure

### first attempt

conf file mounted into `/bitnami/redis/conf/redis.conf`
```
# docker-compose.yml
version: '3'

services: 
  redis:
    image: bitnami/redis
    volumes:
      - ./rediscache.conf:/bitnami/redis/conf/redis.conf
    environment:
      - REDIS_PASSWORD=redis
```

Result:
```
# docker-compose up
redis_1  | 
redis_1  | Welcome to the Bitnami redis container
redis_1  | Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-redis
redis_1  | Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-redis/issues
redis_1  | Send us your feedback at containers@bitnami.com
redis_1  | 
redis_1  | nami    INFO  Initializing redis
redis_1  | redis   INFO  ==> Validating inputs...
redis_1  | Error executing 'postInstallation': EACCES: permission denied, open '/opt/bitnami/redis/conf/redis.conf'
dockerrediscache_redis_1 exited with code 1
```

### second attempt

Conf file added to the image in advance.

```
# Dockerfile
FROM bitnami/redis:4.0
USER root
ADD ./rediscache.conf /bitnami/redis/conf/redis.conf
USER 1001
```

```
# docker-compose.yml
version: '3'

services: 
  redis:
    build: .
    environment:
      - REDIS_PASSWORD=redis
```

```
# docker-compose up --build
redis_1  | 
redis_1  | Welcome to the Bitnami redis container
redis_1  | Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-redis
redis_1  | Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-redis/issues
redis_1  | Send us your feedback at containers@bitnami.com
redis_1  | 
redis_1  | nami    INFO  Initializing redis
redis_1  | Error executing 'postInstallation': ELOOP: too many symbolic links encountered, lstat '/opt/bitnami/redis/conf/redis.conf'
```

I also had a third try and got some messages from nami_initialize about variables that could not be replaced, but I don't remember what I did to get this.

Thanks for your help
Regards

Seyhbold

**Comment by tompizmor** [comment]
Hi @seyhbold,

This is just a quick note to let you know that we could reproduce the permission issue mounting the config file. We will work on a fix and release a new revision of the Redis image.

Regards,
Tomas

**Comment by tompizmor** [comment]
Hi again @seyhbold,

I just released redis `4.0.6-r1`. 

You will be able to mount your config file as you tried on your first attempt and you can also specify extra flags using the REDIS_EXTRA_FLAGS environment variable as documented [here](https://github.com/bitnami/bitnami-docker-redis#passing-extra-command-line-flags-to-redis-server-startup)

I hope this helps,
Tomas

**Comment by seyhbold** [comment]
Hi Tomas,

Thank you very much. I won‘t be able to test this year. But it will be my first activity in january. 

I wish you nice holidays. 
Cheers Seyhbold
